build-system/eslint-rules/no-duplicate-name-typedef.js

```diff
@@ -15,12 +15,19 @@
  */
 'use strict';
 
+const argv = require('minimist')(process.argv.slice(2));
+
 // Global cache of typedefName: typedefLocation.
 const typedefs = new Map();
```

Comment:
I think we can just move this into the `module.exports` function.

Reply:
Tried that first, but `typedefs` gets reset after each file, which defeats the purpose of this check.

Reply:
IIUC, `typedefs` needs to be a global var for this check to work correctly during `gulp lint`, and global vars are not reset by IDE `eslint` plugins.

In any case, I don't think it makes sense to apply a rule that spans all files while linting a single file.